### PR TITLE
Normalization crazyness

### PR DESCRIPTION
**Pull request by leogrim**
@yingjieMiao @raymondng 
Please review carefully, mainly the last two commits, the rest is cosmetic :)

### CODE REVIEW

File: kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala [hunk 1]
```diff
@@ -375,24 +375,39 @@ class ScrapeWorker(
...
-      case Some(redirect) if !redirect.isPermanent || hasFishy301(uri) => {
+    redirects.dropWhile(!_.isLocatedAt(uri.url)) match {
+      case Seq(redirect, _*) if !redirect.isPermanent || hasFishy301(uri) => {
         if (redirect.isPermanent) log.warn(s"Found fishy $redirect for $uri") else log.warn(s"Found non permanent $redirect for $uri")
         updateRedirectRestriction(uri, redirect)
       }
-      case Some(permanentRedirect) if permanentRedirect.isAbsolute => {
-        log.debug(s"Found permanent $permanentRedirect for $uri")
-        helper.syncRecordPermanentRedirect(removeRedirectRestriction(uri), permanentRedirect)
+      case permanentsRedirects => {
```
Comment (raymondng) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:389]: As discussed, would be very nice to have a test to cover the multi-redirect scenario

**Comment by raymondng** [comment]
Thanks @leogrim!

File: kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala [hunk 2]
```diff
@@ -375,24 +375,39 @@ class ScrapeWorker(
...
     httpContentType = extractor.getMetadata("Content-Type"),
     httpOriginalContentCharset = extractor.getMetadata("Content-Encoding"),
-    destinationUrl = Some(destinationUrl),
+    destinationUrl = destinationUrl,
     signature = getSignature(extractor)
   )
 
   // Watch out: the NormalizedURI may come back as REDIRECTED
   private def processRedirects(uri: NormalizedURI, redirects: Seq[HttpRedirect]): NormalizedURI = {
-    redirects.find(_.isLocatedAt(uri.url)) match {
```
Comment (yingjieMiao) [inline kifi-backend/modules/scraper/app/com/keepit/scraper/ScrapeWorker.scala:384]: it would be nice to document what this function does.

**Comment by yingjieMiao** [comment]
awesome.

**Pull request merged**